docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,216 @@
+
+### Controlling rollout
+
+To begin, ensure all component services are up and running, and the working directory has a `.env` file containing the appropriate `ENGINE_API_KEY` for your federated service. Then, upload each service to the gateway through `Apollo CLI`:
+
+```bash
+apollo service:push       \
+  --serviceURL={{url}}    \
+  --endpoint={{url}}      \
+  --serviceName={{name}}  \
+  --federated
+```
+
+> Each service needs a unique name, this is how you will identify the service for schema updates
+
+The given `endpoint` will be queried for the new schema which, if compatible with the rest of the schema, will be uploaded to `Apollo Graph Manager`. At this point, any gateways serving this graph will begin to roll over to the new schema.
+
+You can later modify hosted service definitions by rerunning the `apollo service:push` command with the service's name and the new URL.
+
+> What's the difference between `serviceURL` and `endpoint` parameters? The `endpoint` parameter controls the endpoint where the schema will be fetched from at composition, whereas `serviceURL` controls what URL the gateway will query at runtime. This is especially helpful if you have disabled introspection queries in your production services.
+
+// TODO: Document the integration with service:push, services under a graph, and gateway lifecycle management utilities
+
+### Reliability
+
+[//]: # (Description: This section should document how the gateway polls GCS for updates, why it's a reliable model, what the defaults are, and any recommendations)
+[//]: # (Assignee: Jackson)
+[//]: # (Reviewer: Adam)
+
+When operating as a managed gateway, `Apollo Server` will poll the [Schema Registry](https://www.apollographql.com/docs/platform/schema-registry/) for updates to the registered service list at 10 second intervals. When a service list update does occur, `Apollo Server` will create a new composed schema from the federated services, and begin to roll over to serving the new schema. Existing in-flight operations on the old schema will continue to be processed while serving the new schema. For this reason, it can be helpful to serve the new schemas from new endpoints, such that no downtime is incurred during rollover.
+
+In the event a network failure prevents an `Apollo Server` gateway from contacting the Schema Registry, the gateway will continue to serve the last known schema while it attempts to reestablish a connection to the registry.
+
+<!--
+  Jackson: There is no way of falling back to local files/introspection currently, but I also don't think there should be, consider:
```

Comment:
I think there should certainly be a way of falling back to local files. GCS being down shouldn't mean that the entirety of your graph is down.